examples/distributed/graphlearn_for_pytorch/partition_ogbn_dataset.py

```diff
@@ -0,0 +1,116 @@
+import argparse
+import ast
+import os.path as osp
+
+import graphlearn_torch as glt
+import torch
+from ogb.nodeproppred import PygNodePropPredDataset
+
+
+def partition_dataset(ogbn_dataset: str, root_dir: str, num_partitions: int,
+                      num_nbrs: glt.NumNeighbors, chunk_size: int,
+                      cache_ratio: float):
+    print(f'-- Loading {ogbn_dataset} ...')
+    dataset = PygNodePropPredDataset(ogbn_dataset, root_dir)
+    data = dataset[0]
+    node_num = len(data.x)
+    edge_num = len(data.edge_index[0])
+    print('* node count: {}'.format(node_num))
+    print('* edge count: {}'.format(edge_num))
+    split_idx = dataset.get_idx_split()
+
+    print('-- Saving label ...')
+    label_dir = osp.join(root_dir, f'{ogbn_dataset}-label')
+    glt.utils.ensure_dir(label_dir)
+    torch.save(data.y.squeeze(), osp.join(label_dir, 'label.pt'))
+
+    print('-- Partitioning training idx ...')
+    train_idx = split_idx['train']
+    train_idx = train_idx.split(train_idx.size(0) // num_partitions)
+    train_idx_partitions_dir = osp.join(root_dir,
+                                        f'{ogbn_dataset}-train-partitions')
+    glt.utils.ensure_dir(train_idx_partitions_dir)
+    for pidx in range(num_partitions):
+        torch.save(train_idx[pidx],
+                   osp.join(train_idx_partitions_dir, f'partition{pidx}.pt'))
+
+    print('-- Partitioning test idx ...')
+    test_idx = split_idx['test']
+    test_idx = test_idx.split(test_idx.size(0) // num_partitions)
+    test_idx_partitions_dir = osp.join(root_dir,
+                                       f'{ogbn_dataset}-test-partitions')
+    glt.utils.ensure_dir(test_idx_partitions_dir)
+    for pidx in range(num_partitions):
+        torch.save(test_idx[pidx],
+                   osp.join(test_idx_partitions_dir, f'partition{pidx}.pt'))
+
+    print('-- Initializing graph ...')
+    csr_topo = glt.data.CSRTopo(edge_index=data.edge_index, layout='COO')
+    graph = glt.data.Graph(csr_topo, mode='ZERO_COPY')
+
+    print('-- Sampling hotness ...')
+    glt_sampler = glt.sampler.NeighborSampler(graph, num_nbrs)
+    node_probs = []
+    for pidx in range(num_partitions):
+        seeds = train_idx[pidx]
+        prob = glt_sampler.sample_prob(seeds, node_num)
+        node_probs.append(prob.cpu())
```

Comment:
Curious, is this step required for the partitioner?

Reply:
This step is calculating node access distribution for cache and faster sampling, and it is necessary for FrequencyPartitioner here. 
We also provide the random partitioner that directly partitions randomly.